lib/linguist/heuristics.yml

```diff
@@ -672,6 +672,10 @@ disambiguations:
   rules:
   - language: STL
     pattern: '\A\s*solid(?=$|\s)(?m:.*?)\Rendsolid(?:$|\s)'
+- extensions: ['.sw']
+  rules:
+  - language: Sway
+    pattern: '^(dep |fn |mod |pub |trait |impl |abi #!?\[)'
```

Comment:
I think this can definitely be optimised, especially the final `abi #!?\[` part as I don't think this will match what I think you're hoping it'll match based on the samples included.

/cc @Alhadis as the resident regex guru.

Reply:
This regex will match any of the given strings:

1. `dep `
2. `fn `
3. `mod `
4. `pub `
5. `trait `
6. `impl `
7. `abi #[`
8. `abi #![`

I don't see any `#` characters in the samples, but without making assumptions about author intent, a more robust regex would be

~~~regexp
^\s*(?:(?:dep|fn|impl|mod|pub|trait)\s|abi\s#!?\[)
~~~

Reply:
Updated it, I'm expecting it to support:
1. `dep` 
1. `fn`
1. `mod` 
1. `pub` 
1. `trait` 
1. `impl` 
1. `abi`
1. `#[`

Thanks for pointing this out!

Reply:
Updated the samples as well. Let me know what you think.

Reply:
> I'm expecting it to support:

Is this is an exclusive list? As in, `abi` should be matched separately to `#[`? If so, then a better regex would be

~~~regexp
^\s*(?:(?:abi|dep|fn|impl|mod|pub|trait)\s|#\[)
~~~